cra-to-nx fails

## Current Behavior
Running a `npx cra-to-nx` on cra projects fails in an error.
Tried it on a freshly generated cra from today and a two year old code with react 17.

## Expected Behavior
`cra-to-nx` should work and migrate the code to an nx monorepo style with a react app.

## Steps to Reproduce
1. Create a fresh [cra](https://create-react-app.dev/docs/getting-started/).
2. Run `npx cra-to-nx`
3. Failiure


### Failure Logs

```
npm ERR! code ENOVERSIONS
npm ERR! No valid versions available for temp-workspace

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/slavoroikhel/.npm/_logs/2022-08-17T15_39_28_845Z-debug.log
Install for [ 'temp-workspace@latest' ] failed with code 1
Error: Command failed: npx -y create-nx-workspace@latest temp-workspace --appName=cra-test --preset=react --style=css --packageManager=npm
    at checkExecSyncError (node:child_process:828:11)
    at execSync (node:child_process:899:15)
    at /Users/slavoroikhel/.npm/_npx/18107/lib/node_modules/cra-to-nx/src/lib/cra-to-nx.js:52:38
    at Generator.next (<anonymous>)
    at /Users/slavoroikhel/.npm/_npx/18107/lib/node_modules/cra-to-nx/node_modules/tslib/tslib.js:118:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/Users/slavoroikhel/.npm/_npx/18107/lib/node_modules/cra-to-nx/node_modules/tslib/tslib.js:114:16)
    at createNxWorkspaceForReact (/Users/slavoroikhel/.npm/_npx/18107/lib/node_modules/cra-to-nx/src/lib/cra-to-nx.js:40:20)
    at Object.<anonymous> (/Users/slavoroikhel/.npm/_npx/18107/lib/node_modules/cra-to-nx/src/index.js:9:43) {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 18147,
  stdout: null,
  stderr: null
}
```

_**_The log file:_**_ 
```
0 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   '/Users/slavoroikhel/.nvm/versions/node/v16.15.0/bin/node',
1 verbose cli   '/Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/bin/npm-cli.js',
1 verbose cli   'install',
1 verbose cli   'temp-workspace@latest',
1 verbose cli   '--global',
1 verbose cli   '--prefix',
1 verbose cli   '/Users/slavoroikhel/.npm/_npx/15842',
1 verbose cli   '--loglevel',
1 verbose cli   'error',
1 verbose cli   '--json'
1 verbose cli ]
2 info using npm@6.14.17
3 info using node@v16.15.0
4 verbose npm-session d6606aedd39144cd
5 silly install loadCurrentTree
6 silly install readGlobalPackageData
7 http fetch GET 304 https://registry.npmjs.org/temp-workspace 846ms (from cache)
8 silly fetchPackageMetaData error for temp-workspace@latest No valid versions available for temp-workspace
9 timing stage:rollbackFailedOptional Completed in 0ms
10 timing stage:runTopLevelLifecycles Completed in 855ms
11 verbose type tag
12 verbose stack temp-workspace: No valid versions available for temp-workspace
12 verbose stack     at pickManifest (/Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/npm-pick-manifest/index.js:39:11)
12 verbose stack     at /Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/pacote/lib/fetchers/registry/manifest.js:24:14
12 verbose stack     at tryCatcher (/Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
12 verbose stack     at Promise._settlePromiseFromHandler (/Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:517:31)
12 verbose stack     at Promise._settlePromise (/Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:574:18)
12 verbose stack     at Promise._settlePromise0 (/Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:619:10)
12 verbose stack     at Promise._settlePromises (/Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:699:18)
12 verbose stack     at _drainQueueStep (/Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:138:12)
12 verbose stack     at _drainQueue (/Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:131:9)
12 verbose stack     at Async._drainQueues (/Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:147:5)
12 verbose stack     at Immediate.Async.drainQueues [as _onImmediate] (/Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:17:14)
12 verbose stack     at processImmediate (node:internal/timers:466:21)
13 verbose cwd /Users/slavoroikhel/dev/playground/cra-test
14 verbose Darwin 21.6.0
15 verbose argv "/Users/slavoroikhel/.nvm/versions/node/v16.15.0/bin/node" "/Users/slavoroikhel/.nvm/versions/node/v16.15.0/lib/node_modules/npm/bin/npm-cli.js" "install" "temp-workspace@latest" "--global" "--prefix" "/Users/slavoroikhel/.npm/_npx/15842" "--loglevel" "error" "--json"
16 verbose node v16.15.0
17 verbose npm  v6.14.17
18 error code ENOVERSIONS
19 error No valid versions available for temp-workspace
20 verbose exit [ 1, true ]
```

### Environment
Mac M1
node 16.15.0
npm 6.15.17

### Additional comments
Sounds like arguments issue with
`http fetch GET 304 https://registry.npmjs.org/temp-workspace`

--- a/packages/cra-to-nx/src/lib/cra-to-nx.ts
+++ b/packages/cra-to-nx/src/lib/cra-to-nx.ts
@@ -9,6 +9,7 @@ import {
   removeSync,
   readdirSync,
 } from 'fs-extra';
+import { output } from '@nrwl/devkit';
 
 import { addCRAcracoScriptsToPackageJson } from './add-cra-commands-to-nx';
 import { checkForUncommittedChanges } from './check-for-uncommitted-changes';
@@ -17,7 +18,6 @@ import { readNameFromPackageJson } from './read-name-from-package-json';
 import { setupTsConfig } from './tsconfig-setup';
 import { writeCracoConfig } from './write-craco-config';
 import { cleanUpFiles } from './clean-up-files';
-import { output } from '@nrwl/devkit';
 
 let packageManager: string;
 function checkPackageManager() {
@@ -60,9 +60,14 @@ export async function createNxWorkspaceForReact(options: Record<string, any>) {
     ...packageJson.devDependencies,
   };
   const isCRA5 = /^[^~]?5/.test(deps['react-scripts']);
+  const npmVersion = execSync('npm -v').toString();
+  // Should remove this check 04/2023 once Node 14 & npm 6 reach EOL
+  const npxYesFlagNeeded = !npmVersion.startsWith('6'); // npm 7 added -y flag to npx
 
   execSync(
-    `npx -y create-nx-workspace@latest temp-workspace --appName=${reactAppName} --preset=react --style=css --packageManager=${packageManager}`,
+    `npx ${
+      npxYesFlagNeeded ? '-y' : ''
+    } create-nx-workspace@latest temp-workspace --appName=${reactAppName} --preset=react --style=css --packageManager=${packageManager}`,
     { stdio: [0, 1, 2] }
   );
 

